### ISSUE TITLE
[MU4 Issue] Many dialog windows have missing app icon in the title bar on Windows

### ISSUE THREAD

**Issue by DmitryArefiev** [musescore/MuseScore#10597]
**Describe the bug**
Many dialog windows have missing app icon in title bar on Windows:
- "Do you want to save changes.." dialog
- New Score
- Parts
- Instruments
- Export
- Preferences
- Workspaces
- Create new workspaces
- Insert measures
- About MuseScore
- About MusicXML
- Check for update
- Revert to factory settings
- All dialogs from Diagnostic tab
- Migration dialog (import of previous version scores) 
- Error message dialog when trying to add text to a not single selected note
(Maybe I missed some dialogs..)

![MU4](https://user-images.githubusercontent.com/90187801/154470643-f482077b-3ddf-4e1d-aa46-8ab5940ee002.PNG)
![MU4 1](https://user-images.githubusercontent.com/90187801/154470654-229a7b05-9249-4a71-8f57-1b8bd76ee217.PNG)
![MU4 2](https://user-images.githubusercontent.com/90187801/154470666-a399e9d1-9df7-4dbf-8c62-2bd884a2eb4d.PNG)

But some dialogs have app icon:
![MU4 5](https://user-images.githubusercontent.com/90187801/154470679-21b02c8d-0392-4a20-829e-e3bc98be5087.PNG)

**Platform information**
 Windows (Win10)

**Additional context**
Doesn't occur in **MU3**

**Comment by DmitryArefiev** [comment]
@Tantacrul Maybe to put this to Beta1 but with lower priority? Since it occurs in many windows (on Windows))

**Comment by Tantacrul** [comment]
As annoying as it is, I can't put it in Beta 1 when there are numerous score / MIDI / audio issues to solve. Let's just wait until all the Muse Hub, Muse Sounds and MuseScore.com upload bugs start coming in!

**Comment by DmitryArefiev** [comment]
Sure! I understand.

**Comment by RmGkAppDev** [comment]
I wish the title bar would have been custom-made - like the one on the main window where you don't have this screaming bright line that is burning my eyes every time I look at it. (By the way -  Microsoft made a terrible design choice by making the title bar white)

**Comment by cbjeukendrup** [comment]
I wish it _could_ have been custom made! Unless I'm completely missing something, it is so terribly difficult to simply change the color of the window title bar on Windows. I have spent hours in my life just googling how to change the title bar color, but it seems simply not doable in a way that works together with the things that we use (c++, Qt) (without doing crazily complicated stuff that is not worth the effort).

(FWIW, I do think that you can change the title bar color on your own computer. Windows has a sort of dark mode feature, somewhere hidden in Settings.)

**Comment by RmGkAppDev** [comment]
> I wish it _could_ have been custom made! Unless I'm completely missing something, it is so terribly difficult to simply change the color of the window title bar on Windows. I have spent hours in my life just googling how to change the title bar color, but it seems simply not doable in a way that works together with the things that we use (c++, Qt) (without doing crazily complicated stuff that is not worth the effort).
> 
> (FWIW, I do think that you can change the title bar color on your own computer. Windows has a sort of dark mode feature, somewhere hidden in Settings.)

Maybe this will help:
https://stackoverflow.com/questions/30181630/qdialog-remove-title-bar
It's not much, but if you can hide the title bar (at least in a dialog), you might be able to add your own custom draggable bar with a header and an X button.
I've done this a few times before - but only in WPF and Electron, never in Qt (never programmed in Qt...)
It took me a bit of effort, and I believe it's not that bad in Qt either.

**Comment by Tantacrul** [comment]
Yeah, I'd opt in future for us completely replacing the popup so we can use our own custom header (a bit like we do with the main title bar).

**Comment by papoteur-mga** [comment]
Hello,
I have a similar behaviour in Linux (Mageia), but on the main window. The icon at top left is an X like generic application. 
The issue is also for the taskbar in LXQt, but not in Plasma.
The icon is well displayed in menu.
A friend reported me that the top left is also an X like generic application with the appimage.

**Comment by papoteur-mga** [comment]
I applied a correction which allows to get the icon in the main window:
adding this line:
```
    app.setWindowIcon(QIcon::fromTheme(QStringLiteral("mscore")));
```
at line 112:
https://github.com/musescore/MuseScore/blob/3969d7975534927ab859c154fcee9a47b0acfe2b/src/appshell/appshell.cpp#L111

**Comment by cbjeukendrup** [comment]
@papoteur-mga Nice! Would you like to open a Pull Request for this yourself?

**Comment by papoteur-mga** [comment]
If you can do it, I would be happy.
Just a line to add, but I'm not comfortable with PR.

**Comment by cbjeukendrup** [comment]
@papoteur-mga I tried it out, and unfortunately this doesn't work on Windows... I suspect `QIcon::fromTheme` is very platform dependent. So we'll need to look further...
